### PR TITLE
Add support for AWS SDK 2.0 for DynamoDb

### PR DESCRIPTION
**Pull request by tyiu**
Misk currently supports only AWS SDK 1.11. This change does a few things:
1. Splits out DynamoDb (AWS SDK 1.11) code out from `misk-aws` into its own `misk-aws-dynamodb` module.
2. Creates new `misk-aws2-dynamodb` and `misk-aws2-dynamodb-testing` modules, backed by AWS SDK 2.0 under a separate `misk.aws2` package.

See the SDK homepage for details on what AWS SDK 2.0 provides:
https://docs.aws.amazon.com/sdk-for-java/latest/migration-guide/what-is-java-migration.html

### CODE REVIEW

**Comment by tyiu** [comment]
Most of the files in the `misk-aws2-dynamodb-testing/src/main/kotlin/misk/aws2/dynamodb/testing/` path were taken from https://github.com/cashapp/tempest/tree/e4ee11cb23682b2e60937f7b5965e7e78b28d93c/tempest2/src/test/kotlin/misk/aws2/dynamodb/testing thanks to @zhxnlai.

File: misk-aws2-dynamodb-testing/src/main/kotlin/misk/aws2/dynamodb/testing/InProcessDynamoDbModule.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    install(LocalDynamoDbModule(tables))
+    install(ServiceModule<InProcessDynamoDbService>())
+    install(ServiceModule<CreateTablesService>()
+        .dependsOn(InProcessDynamoDbService::class.toKey()))
+  }
+
+  @Provides @Singleton
+  internal fun provideDynamoDBProxyServer(localDynamoDb: LocalDynamoDb): DynamoDBProxyServer {
+    Libsqlite4JavaLibraryPathInitializer.init()
+    return ServerRunner.createServerFromCommandLineArgs(
```
Comment (tyiu) [inline misk-aws2-dynamodb-testing/src/main/kotlin/misk/aws2/dynamodb/testing/InProcessDynamoDbModule.kt:35]: Inspired by https://github.com/aws/aws-sdk-java-v2/blob/46f6252baf1842b85723c27835558977ca39a4c8/services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/LocalDynamoDb.java

**Review by zhxnlai — APPROVED** [review]

File: misk-aws-dynamodb/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=misk-aws-dynamodb
```
Comment (zhxnlai) [inline misk-aws-dynamodb/gradle.properties:1]: do you have plans to migrate the existing usages?

File: misk-aws-dynamodb/build.gradle [hunk 3]
```diff
@@ -0,0 +1,15 @@
+dependencies {
+  implementation dep.guice
+  implementation dep.awsDynamodb
+  implementation project(':misk-aws')
```
Comment (zhxnlai) [inline misk-aws-dynamodb/build.gradle:4]: is this needed?

Comment (tyiu) [reply misk-aws-dynamodb/build.gradle:4]: Yes. The `AwsRegion` class is needed from that package.

File: misk-aws-dynamodb/gradle.properties (hunk 2 above)
Comment (tyiu) [reply misk-aws-dynamodb/gradle.properties:1]: Yes.

File: misk-aws2-dynamodb-testing/gradle.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+POM_ARTIFACT_ID=misk-aws-dynamodb-testing
```
Comment (swankjesse) [inline misk-aws2-dynamodb-testing/gradle.properties:1]: aws2?

File: misk-aws2-dynamodb-testing/src/test/kotlin/misk/aws2/dynamodb/testing/AbstractDynamoDbTest.kt [hunk 5]
```diff
@@ -0,0 +1,143 @@
+package misk.aws2.dynamodb.testing
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient
```
Comment (swankjesse) [inline misk-aws2-dynamodb-testing/src/test/kotlin/misk/aws2/dynamodb/testing/AbstractDynamoDbTest.kt:5]: does enhanced mean v2?

**Review by swankjesse — APPROVED** [review]
Is it safe to install both side-by-side? Should we offer guidance?

File: misk-aws2-dynamodb-testing/gradle.properties (hunk 4 above)
Comment (tyiu) [reply misk-aws2-dynamodb-testing/gradle.properties:1]: Yep. Typo.

File: misk-aws2-dynamodb-testing/src/test/kotlin/misk/aws2/dynamodb/testing/AbstractDynamoDbTest.kt (hunk 5 above)
Comment (tyiu) [reply misk-aws2-dynamodb-testing/src/test/kotlin/misk/aws2/dynamodb/testing/AbstractDynamoDbTest.kt:5]: There's also a non-enhanced client that's also in v2. Anything in the software.amazon.awssdk.* package is v2.

**Comment by tyiu** [comment]
> Is it safe to install both side-by-side? Should we offer guidance?

@swankjesse Yes and yes. I'll follow-up with documentation.

**Pull request merged**